packages/spindle-ui/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback } from 'react';
+import MenuHorizontal from '../Icon/MenuHorizontal';
+
+import PaginationItem from './PaginationItem';
+import { useShowItem } from './hooks/useShowItem';
+
+interface Props extends React.HTMLAttributes<HTMLElement> {
+  current: number;
+  total: number;
+  showCount?: boolean;
+  showPrevNext?: boolean;
+  showFirstLast?: boolean;
+  onPageChange: (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    pageNumber: number,
+  ) => void;
+  createUrl: (pageNumber: number) => string;
+}
+
+const BLOCK_NAME = 'spui-Pagination';
+
+export const Pagination = (props: Props) => {
+  const {
+    current,
+    total,
+    showCount = false,
+    showPrevNext = true,
+    showFirstLast = false,
+    onPageChange,
+    createUrl,
+    className,
+    ...rest
+  } = props;
+
+  const pageItem = 5;
+  const { showItem, showPrevHorizontal, showNextHorizontal } = useShowItem({
+    current,
+    total,
+    pageItem,
+  });
+
+  const handleOnClick = useCallback(
+    (
+      event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+      pageNumber: number,
+    ) => {
+      onPageChange?.(event, pageNumber);
+    },
+    [onPageChange],
+  );
+
+  return (
+    <nav
+      aria-label="ページネーション"
+      className={[BLOCK_NAME, className].filter(Boolean).join(' ').trim()}
+      {...rest}
+    >
+      <ul className={`${BLOCK_NAME}-list`}>
+        {showFirstLast && (
+          <li
+            className={[`${BLOCK_NAME}-item`, `${BLOCK_NAME}-item--first`]
+              .join(' ')
+              .trim()}
```

Comment:
```suggestion
            className={`${BLOCK_NAME}-item ${BLOCK_NAME}-item--first`}>
```
条件分岐があるclassNameの場合に`filter(Boolean)`を通すために配列を用いていたかと思うのですが、そのようなパターンでなければ`join`を使わず、シンプルに文字列で良さそうと思いました（ほかの箇所も同様に）🙏